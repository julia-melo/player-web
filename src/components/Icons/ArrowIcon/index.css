.arrow-up-icon svg,
.arrow-down-icon svg,
.arrow-left-icon svg,
.arrow-right-icon svg {
  display: block;
}

.arrow-up-icon {
  transform: rotate(270deg);
}

.arrow-down-icon {
  transform: rotate(90deg);
}

.arrow-left-icon {
  transform: rotate(180deg);
}

.arrow-up-icon.clickable,
.arrow-down-icon.clickable,
.arrow-left-icon.clickable,
.arrow-right-icon.clickable {
  cursor: pointer
}

.arrow-up-icon.disabled,
.arrow-down-icon.disabled,
.arrow-left-icon.disabled,
.arrow-right-icon.disabled {
  opacity: 0.4;
  cursor: auto
}
