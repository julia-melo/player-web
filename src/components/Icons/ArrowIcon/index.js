import React from 'react'
import PropTypes from 'prop-types'
import { ReactSVG } from 'react-svg'

import arrowIcon from '../../../assets/arrow.svg'

import './index.css'

const ArrowIcon = ({
  className,
  width = 80,
  height = 80,
  disabled = false,
  onClick
}) => (
  <ReactSVG
    className={`${className} ${disabled ? 'disabled' : ''} ${onClick ? 'clickable' : ''}`}
    src={arrowIcon}
    beforeInjection={svg => {
      svg.setAttribute('style', `width: ${width}px; height: ${height}px;`)
    }}
    onClick={onClick}
  />
)

ArrowIcon.propTypes = {
  className: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
}

const ArrowRightIcon = props => (
  <ArrowIcon className='arrow-right-icon' {...props} />
)

const ArrowLeftIcon = props => (
  <ArrowIcon className='arrow-left-icon' {...props} />
)

const ArrowUpIcon = props => (
  <ArrowIcon className='arrow-up-icon' {...props} />
)

const ArrowDownIcon = props => (
  <ArrowIcon className='arrow-down-icon' {...props} />
)

export { ArrowUpIcon, ArrowDownIcon, ArrowLeftIcon, ArrowRightIcon }
