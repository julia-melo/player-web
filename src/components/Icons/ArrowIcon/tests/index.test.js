import React from 'react'
import renderer from 'react-test-renderer'
import { ReactSVG } from 'react-svg'

import ArrowIcon, {
  ArrowRightIcon,
  ArrowLeftIcon,
  ArrowUpIcon,
  ArrowDownIcon
} from '../index'

describe('ArrowIcon component', () => {
  it('fires svg beforeInjection', () => {
    const instance = renderer
      .create(<ArrowIcon className='' />)
      .root

    const setAttribute = jest.fn()
    const fakeSVG = { setAttribute }
    instance.findByType(ReactSVG).props.beforeInjection(fakeSVG)

    expect(setAttribute).toHaveBeenCalled()
  })

  it('receives disabled prop', () => {
    const instance = renderer
      .create(<ArrowIcon className='' disabled />)
      .root

    expect(instance.findByType(ReactSVG).props.className).toContain('disabled')
  })

  it('receives onClick prop', () => {
    const clickMock = jest.fn()
    const instance = renderer
      .create(<ArrowIcon className='' onClick={clickMock} />)
      .root

    expect(instance.findByType(ReactSVG).props.className).toContain('clickable')
  })
})

describe('ArrowRightIcon component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ArrowRightIcon />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ArrowLeftIcon component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ArrowLeftIcon />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ArrowUpIcon component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ArrowUpIcon />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ArrowDownIcon component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ArrowDownIcon />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
