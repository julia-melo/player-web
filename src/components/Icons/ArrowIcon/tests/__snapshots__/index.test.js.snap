// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowDownIcon component renders correctly 1`] = `
<div
  className="arrow-down-icon  "
/>
`;

exports[`ArrowLeftIcon component renders correctly 1`] = `
<div
  className="arrow-left-icon  "
/>
`;

exports[`ArrowRightIcon component renders correctly 1`] = `
<div
  className="arrow-right-icon  "
/>
`;

exports[`ArrowUpIcon component renders correctly 1`] = `
<div
  className="arrow-up-icon  "
/>
`;
